src/resampling.jl

```diff
@@ -1,41 +1,60 @@
 ## RESAMPLING STRATEGIES
 
 abstract type ResamplingStrategy <: MLJType end
-import Random
+
 # resampling strategies are `==` if they have the same type and their
 # field values are `==`:
-function ==(s1::S, s2::S) where S<:ResamplingStrategy
-    ret = true
-    for fld in fieldnames(S)
-        ret = ret && getfield(s1, fld) == getfield(s2, fld)
-    end
-    return ret
+function ==(s1::S, s2::S) where S <: ResamplingStrategy
+    return all(getfield(s1, fld) == getfield(s2, fld) for fld in fieldnames(S))
 end
 
-mutable struct Holdout <: ResamplingStrategy
-    fraction_train::Float64
-    shuffle::Bool
-    rng::Union{Int,AbstractRNG}
+"""
+$TYPEDEF
+
+Single train-test split with a (randomly selected) portion of the
+data being selected for training and the rest for testing.
+
+* `fraction_train` a number between 0 and 1 indicating the proportion
+of the samples to use for training
+* `shuffle` a boolean indicating whether to select the training samples
+at random
+* `rng` a random number generator to use
+"""
+@with_kw mutable struct Holdout <: ResamplingStrategy
+    fraction_train::Float64 = 0.7
+    shuffle::Bool = false
+    rng::Union{Int,AbstractRNG} = Random.GLOBAL_RNG
+
     function Holdout(fraction_train, shuffle, rng)
-        0 < fraction_train && fraction_train < 1 ||
-            error("fraction_train must be between 0 and 1.")
+        0 < fraction_train < 1 || error("`fraction_train` must be between 0 and 1.")
         return new(fraction_train, shuffle, rng)
     end
 end
-Holdout(; fraction_train=0.7, shuffle=false, rng=Random.GLOBAL_RNG) = Holdout(fraction_train, shuffle, rng)
+
 show_as_constructed(::Type{<:Holdout}) = true
 
-mutable struct CV <: ResamplingStrategy
-    nfolds::Int
-    parallel::Bool
-    shuffle::Bool ## TODO: add seed/rng
-    rng::Union{Int,AbstractRNG}
+"""
+$TYPEDEF
+
+Cross validation resampling where the data is (randomly) partitioned in `nfolds` folds
+and the model is evaluated `nfolds` time, each time taking one fold for testing and the
+other folds for training and the test  performances averaged.
+For instance if `nfolds=3` then the data will be partitioned in three folds A, B and C
+and the model will be trained three times, first with A and B and tested on C, then on
+A, C and tested on B and finally on B, C and tested on A. The test performances are then
+averaged over the three cases.
+"""
```

Comment:
Whether or not scores are averaged or not depends on the context in which a strategy is used. So, in `evaluate!` all scores are returned (no aggregation). In Grid() tuning, the scores are averaged. But in another tuning strategy (eg, Bayesian optimisation), all scores might get used for updates. 

Maybe a case of premature documentation : -)